packages/transport-ble/src/transports/BleTransportSession.ts

```diff
@@ -0,0 +1,35 @@
+import type { Central } from '@animo-id/react-native-ble-didcomm'
+import type { Agent, EncryptedMessage, TransportSession } from '@aries-framework/core'
+
+import { TransportService, JsonEncoder } from '@aries-framework/core'
+
+export class BleTransportSession implements TransportSession {
+  public readonly type = 'BluetoothLowEnergy'
+  public id: string
+  private agent: Agent
+  private central: Central
+
+  public constructor(id: string, sdk: Central, agent: Agent) {
+    this.id = id
+    this.agent = agent
+    this.central = sdk
+  }
+
+  public async send(encryptedMessage: EncryptedMessage): Promise<void> {
+    const serializedMessage = JsonEncoder.toString(encryptedMessage)
+
+    this.agent.config.logger.debug('Sending BLE inbound message')
+
+    await this.central.sendMessage(serializedMessage)
+  }
+
+  public async close(): Promise<void> {
+    this.agent.config.logger.debug('Stopping BLE inbound transport')
+
+    await this.central.shutdown()
+
+    const transportService = this.agent.dependencyManager.resolve(TransportService)
+
+    transportService.removeSession(this)
```

Comment:
We don't need to remove it from the transport table. If this method is called it won't be added.

It's important to remove it _if_ the sessions has been saved and suddenly the session is terminated. This could be due to bluetooth being turned of, or the other party disconnecting, but in both cases this method won't be called. 

Look at e.g. the WsInboundTransport in the node package. When a websocket is opened in the `this.socketServer.on('connection'` call, it will register a `socket.on('close',` call to make sure to remove it from the transport service if the session is closed outside of AFJ.

We need a way to remove it when a ble connection is broken. Is there an event in the ble sdk that we can listen to? If not it will accumulate sessions over time that it can't send messages to anymore, which could cause issues as afj will try to send messages to the session.

Reply:
I understand now. I don't know if there's an event emitter we can listen to for broken connections on the SDK. I'll ask @blu3beri and we'll cook something up